idea/src/org/jetbrains/kotlin/idea/inspections/RedundantWithInspection.kt

```diff
@@ -86,7 +93,13 @@ private class RemoveRedundantWithFix : LocalQuickFix {
     override fun applyFix(project: Project, descriptor: ProblemDescriptor) {
         val callExpression = descriptor.psiElement.parent as? KtCallExpression ?: return
         val lambdaBody = callExpression.valueArguments.getOrNull(1)?.lambdaExpression()?.bodyExpression ?: return
-        val replaced = callExpression.replaced(lambdaBody)
+        val singleReturnedExpression = (lambdaBody.statements.singleOrNull() as? KtReturnExpression)?.returnedExpression
+        val newElement = if (singleReturnedExpression != null && callExpression.isDeclarationInitializer()) {
+            singleReturnedExpression
+        } else {
+            lambdaBody
+        }
+        val replaced = callExpression.replaced(newElement)
```

Comment:
@t-kameyama would it be also possible to include the case mentioned in `testData/inspectionsLocal/redundantWith/notApplicable_inFunctionBodyWithReturn.kt`?
```
val lambdaExpression = callExpression.valueArguments.getOrNull(1)?.lambdaExpression()
KtPsiUtil.skipSiblingsBackwardByPredicate(callExpression) { element -> element.text != "=" }?.delete()
val replaced = callExpression.replaced(lambdaExpression!!)
```

Reply:
Fixed in e9fb229